### ISSUE TITLE
Extension 'has_coref' already exists on Doc.

### ISSUE THREAD

**Issue by humehta** [huggingface/neuralcoref#57]
My code:

import spacy
import en_coref_sm

nlp = en_coref_sm.load()
doc = nlp(u'The lungs are located in the chest.They are conical in shape.')

print (doc._.has_coref)
print (doc._.coref_clusters)

Hey I ran into the following error when I inputted my own sentence::::

ValueError                                Traceback (most recent call last)
<ipython-input-3-cd9dab4c3fc0> in <module>()
      2 import en_coref_sm
      3 
----> 4 nlp = en_coref_sm.load()
      5 doc = nlp(u'The lungs are located in the chest.They are conical in shape.')
      6 

~\Anaconda3\lib\site-packages\en_coref_sm\__init__.py in load(**overrides)
     13     overrides['disable'] = disable + ['neuralcoref']
     14     nlp = load_model_from_init_py(__file__, **overrides)
---> 15     coref = NeuralCoref(nlp.vocab)
     16     coref.from_disk(nlp.path / 'neuralcoref')
     17     nlp.add_pipe(coref, name='neuralcoref')

neuralcoref.pyx in en_coref_sm.neuralcoref.neuralcoref.NeuralCoref.__init__()

doc.pyx in spacy.tokens.doc.Doc.set_extension()

ValueError: [E090] Extension 'has_coref' already exists on Doc. To overwrite the existing extension, set `force=True` on `Doc.set_extension`.

**Comment by humehta** [comment]
If I restart the or reload the program it works fine for the first time and gives this error the second time I execute it!!

**Comment by thomwolf** [comment]
Indeed, that arise when you do several successive `en_coref_sm.load()` (see my longer discussion in #52 ). You can try to load the model only once and pass around the nlp object to fix it for now.
This case seems to be quite common so I will add something to fix it. I keep you updated.

**Comment by thomwolf** [comment]
Ok, I have updated the models, it should be fixed now. Please try the new model and tell me if it doesn't for you.

**Comment by humehta** [comment]
Okay i'll try it!!!

**Comment by humehta** [comment]
It still gives me the same error when I try executing the second time.

**Comment by thomwolf** [comment]
Hi @humehta, can you execute the following script and tell me what it displays:
```python
import en_coref_sm
with open(en_coref_md.__file__.replace('__init__.py', 'neuralcoref/neuralcoref.pyx')) as f:
    l = list(f.readlines())
print(l[525])
```

If  you have the latest version of neuralcoref installed, it should print something like:
`if not Doc.has_extension('huggingface_neuralcoref'):`

**Comment by humehta** [comment]
Yes it gives me the following line:
if not Doc.has_extension('huggingface_neuralcoref'):

ALso you have imported the wrong model

**Comment by humehta** [comment]
Hey @thomwolf  any updates?

**Comment by thomwolf** [comment]
Well I must say I have no idea what happening on your system. I can't seems to reproduce this bug with the latest version that you seems to have too (the one that says `if not Doc.has_extension('huggingface_neuralcoref'):`).

Can you give me a detailed walk-thought of all the steps to reproduce the bug?

**Comment by humehta** [comment]
I try running it on my jupyter notebook the first time I run and it runs successfully but when I edit the sentence or in fact run it the second time it gives the above error.

**Comment by humehta** [comment]
@thomwolf

**Comment by thomwolf** [comment]
can you show me a picture or a detailed script of what command you are running?

**Comment by thomwolf** [comment]
Oh another idea, maybe try reinstalling a clean version of NeuralCoref without using the cached version by using this command:
```bash
pip install --upgrade --force-reinstall --no-cache-dir MODEL_URL
```

**Comment by thomwolf** [comment]
This should also be fixed in the latest release (4.0).
